kubevirt, ci: add GitHub issue template for flaky tests

dhiller opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as draft: https://github.com/kubevirt/kubevirt/blob/main/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

### What this PR does
Before this PR:
Flaky tests had to be reported with the standard issue templates.

After this PR:
Adds a template with which flaky tests can be reported.

<!-- (optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: -->
Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: <!-- optional: slack, other GH issue, mailinglist, ... -->

### Special notes for your reviewer
/cc @brianmcarey @xpivarc 

<!-- optional -->

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
NONE
```

dhiller commented:
/release-note none

dhiller commented:
/release-note-none

dhiller commented:
This PR is waiting for three weeks now.

@xpivarc may I kindly ask for a review of this one?

brianmcarey commented on .github/ISSUE_TEMPLATE/flaky_test_report.md:
```diff
@@ -0,0 +1,54 @@
+An infra issue is anything "below" the testcase
+
+* Add label commenting `/triage not-reproducible`
+* Move the issue to "Obsolete"
```
Do github issues have an obsolete options?

dhiller replied:
Good point, I think we should change it to just close the issue

```suggestion
* Close the issue, adding a comment with details about the infrastructure issue.
```

WDYT?

brianmcarey replied:
That looks good to me.

brianmcarey commented on .github/ISSUE_TEMPLATE/flaky_test_report.md:
```diff
@@ -0,0 +1,54 @@
+
+## Then decide on one of the following actions:
+
+### **The flake is a critical bug**
```
```suggestion
### **The flake is a bug**
```
I think the actions below should be done for any bug found through a flake.


Also should we have a section for `an issue with the test`  - some flakes have been related to the test not doing what it should - not sure if this would fall under a bug for kubevirt.

dhiller replied:
Good point, I agree!

brianmcarey commented on .github/ISSUE_TEMPLATE/flaky_test_report.md:
```diff
@@ -0,0 +1,54 @@
+
+<!-- labels for flaky tests -->
+/kind flake
+/priority critical-urgent
```
I guess this should only be added if it is actually urgent?

dhiller replied:
I think any flake deserves the urgent label, since it harms our CI.

xpivarc replied:
Isn't this better fit for PR template?

xpivarc replied:
Right I think issues are good for backtracking but the label `/priority critical-urgent` doesn't have  a meaning for issue, correct?

dhiller replied:
For issues this is just a marker and doesn't do anything besides that, you are right. However, I think that this still has meaning.

brianmcarey reviewed (commented):
Looks good @dhiller - I just have a couple of questions inline.

dhiller commented:
> Looks good @dhiller - I just have a couple of questions inline.

@brianmcarey thank you for the review, I addressed your comments. PTAL, thanks a lot!

iholder101 commented on .github/ISSUE_TEMPLATE/flaky_test_report.md:
```diff
@@ -0,0 +1,54 @@
+
+### **The flake is a critical bug**
+
+* Add label commenting `/triage accepted`
```
I guess that a reviewer should assign this label?

dhiller replied:
As stated above, the assignee who executes upon the issue, should add this label. Do you think I should change the wording to make that more explicit?

iholder101 replied:
Seems that I've misunderstood this label's purpose.
Reading its [docs](https://github.com/kubevirt/kubevirt/labels?q=accepted), it says `Indicates an issue or PR is ready to be actively worked on`.

Makes sense to me now, thank you!

iholder101 commented on .github/ISSUE_TEMPLATE/flaky_test_report.md:
```diff
@@ -0,0 +1,54 @@
+* Keep this issue open until the testcase does not fail anymore
+
+### **The flake is non-critical or an issue that is hard to fix**
+* **Quarantine the test** by creating a pull request assigning [QUARANTINE] tag to test name
+* Reference this issue on the PR
```
Should we keep the issue opened to track quarantine tests?

dhiller replied:
No, I don't think that is necessary. We have the report over quarantined tests [1], which we look at regularly.

[1]: https://storage.googleapis.com/kubevirt-prow/reports/quarantined-tests/kubevirt/kubevirt/index.html

iholder101 commented on .github/ISSUE_TEMPLATE/flaky_test_report.md:
```diff
@@ -0,0 +1,54 @@
+* **Quarantine the test** by creating a pull request assigning [QUARANTINE] tag to test name
+* Reference this issue on the PR
+
+### There was an infrastructure issue
+
+An infra issue is anything "below" the testcase
+
+* Add label commenting `/triage not-reproducible`
```
I guess that the fact that the flake wasn't reproducible is not a proof that it's an infrastructure issue, but perhaps the flake occurs rarely. Maybe change the title to "The test is not reproducible"?

dhiller replied:
Hm, good question. Maybe create a new label that better describes this, like `/triage infra-issue` - WDYT?

iholder101 replied:
Sounds good to me!

iholder101 reviewed (commented):
Awesome stuff @dhiller! Thank you!

dhiller commented on .github/ISSUE_TEMPLATE/flaky_test_report.md:
```diff
@@ -0,0 +1,54 @@
+* Keep this issue open until the testcase does not fail anymore
+
+### **The flake is non-critical or an issue that is hard to fix**
+* **Quarantine the test** by creating a pull request assigning [QUARANTINE] tag to test name
```
```suggestion
* **Quarantine the test** by creating a pull request assigning [QUARANTINE] tag to test name and [Quarantine decorator](https://github.com/kubevirt/kubevirt/blob/f85a8117b8a90fd913ec0719faae9506866d1525/tests/decorators/decorators.go#L7) to the test
```

dhiller commented on .github/ISSUE_TEMPLATE/flaky_test_report.md:
```diff
@@ -0,0 +1,54 @@
+
+An infra issue is anything "below" the testcase
+
+* Add label commenting `/triage not-reproducible`
```
```suggestion
* Add label commenting `/triage infra-issue`
```

dhiller commented:
> Required labels detected, running phase 2 presubmits: /test pull-kubevirt-e2e-windows2016 /test pull-kubevirt-e2e-kind-1.27-vgpu /test pull-kubevirt-e2e-kind-sriov /test pull-kubevirt-e2e-k8s-1.30-ipv6-sig-network /test pull-kubevirt-e2e-k8s-1.29-sig-network /test pull-kubevirt-e2e-k8s-1.29-sig-storage /test pull-kubevirt-e2e-k8s-1.29-sig-compute /test pull-kubevirt-e2e-k8s-1.29-sig-operator /test pull-kubevirt-e2e-k8s-1.30-sig-network /test pull-kubevirt-e2e-k8s-1.30-sig-storage /test pull-kubevirt-e2e-k8s-1.30-sig-compute /test pull-kubevirt-e2e-k8s-1.30-sig-operator

I think running CI here is a complete waste :-1:
